tests/post/steps/test_monitoring.py

```diff
@@ -119,10 +129,169 @@ def _check_object_exists():
     utils.retry(_check_object_exists, times=20, wait=3)
 
 
+@given(parsers.parse("the '{ns_name}' namespace exists"))
+def namespace_exists(k8s_client, ns_name):
+    client = k8s_client.resources.get(api_version="v1", kind="Namespace")
+    namespaces = client.get().items
+    try:
+        next(
+            filter(lambda ns: ns.get("metadata", {}).get("name") == ns_name, namespaces)
+        )
+    except StopIteration:
+        try:
+            client.create({"metadata": {"name": ns_name}})
+        except ApiException as err:
+            raise AssertionError(f"Unable to create namespace: {err}")
```

Comment:
Why iterate on all namespaces ?
You can retrieve only one (and create it if it does not exist)
Something like that
```suggestion
    try:
      client.get(name=ns_name)
    except ApiException as err:
      if err.status == 404:
        client.create({"metadata": {"name": ns_name}})
```